spyder/widgets/collectionseditor.py

```diff
@@ -1418,11 +1426,21 @@ def oedit(self, key):
 
     def plot(self, key, funcname):
         """Plot item"""
-        data = self.source_model.get_data()
         import spyder.pyplot as plt
-        plt.figure()
-        getattr(plt, funcname)(data[key])
-        plt.show()
+        data = self.source_model.get_data()
+
+        if self.get_conf('mute_inline_plotting', section='plots'):
+            from IPython.core.pylabtools import print_figure
+            fig, ax = plt.subplots()
+            getattr(ax, funcname)(data[key])
+            png = print_figure(fig)
+            collections_editor_widget = self.parent()
+            collections_editor_widget.sig_show_figure_requested.emit(
+                png, 'image/png')
```

Comment:
I don't understand this: you added `sig_show_figure_requested` to `BaseTableView`, which ins inherited by this class. So, that signal is available here. 

In addition, you're connecting that signal to the one you declared in `CollectionsEditorWidget`. So, you should be able to write `self.sig_show_figure_requested.emit(png, 'image/png')` instead, right?

Reply:
You're absolutely right, I am changing this now. I added the signal to `BaseTableView` after I wrote this code and I did not realize that this allowed me to do the simplification.